src/main/java/org/breedinginsight/services/validators/TraitValidator.java

```diff
@@ -16,70 +16,269 @@
  */
 package org.breedinginsight.services.validators;
 
+import io.micronaut.http.HttpStatus;
+import org.breedinginsight.api.model.v1.response.ValidationError;
+import org.breedinginsight.api.model.v1.response.ValidationErrors;
 import org.breedinginsight.dao.db.enums.DataType;
+import org.breedinginsight.daos.TraitDAO;
 import org.breedinginsight.model.Method;
 import org.breedinginsight.model.Scale;
 import org.breedinginsight.model.Trait;
-import org.breedinginsight.services.exceptions.UnprocessableEntityException;
+
+import javax.inject.Inject;
+import java.util.*;
+import java.util.stream.Collectors;
 
 import static org.apache.commons.lang3.StringUtils.isBlank;
 
 public class TraitValidator {
 
-    public static void checkRequiredTraitFields(Trait trait) throws UnprocessableEntityException {
+    private TraitDAO traitDAO;
 
-        Method method = trait.getMethod();
-        Scale scale = trait.getScale();
+    @Inject
+    public TraitValidator(TraitDAO traitDAO){
+        this.traitDAO = traitDAO;
+    }
```

Comment:
Looking at this code, I would have expected that the static methods that use the `traitDAO` wouldn't work unless the TraitValidator is injected into the class that's using it.

Reply:
Not sure I'm following on this one. None of the static methods use traitDao. Maybe we can chat about this so you can show me what you are referencing?